input/input.go

```diff
@@ -67,15 +67,18 @@ func (in DefaultHandler) Process(metric *schema.MetricData, partition int32) {
 		return
 	}
 
+	schemaI, _ := mdata.MatchSchema(metric.Name)
```

Comment:
2 pattern matches actually. one for schema, once for aggregation.
that's a good idea. So basically instead of *passing in* the schemaI and aggI into AddOrUpdate, AddOrUpdate would *return* those values instead?
That seems like a good idea to me. For now I've purposely left all the matching obviously unoptimized, to get an idea of how it would perform, but you're right that it'll likely be quite the impact. I still have to measure it though.